### PR TITLE
Arnold/suspension ttl

### PR DESCRIPTION
**Pull request by arnoldyahad**
also added the function to see the response body back

### CODE REVIEW

File: spotinst-cli [hunk 1]
```diff
@@ -367,23 +370,19 @@ for group, group_values in groups_to_update.iteritems():
         req_path = "{}/{}/suspension".format(base_path, group_id)
 
         if options.suspend is not None:
-            if options.suspend != 'BOTH':
-                req_type = 'POST'
-                command = "suspend"
-                payload = '{"processes":["%s"]}' % options.suspend
+            req_type = 'POST'
```
Comment (orarnon) [inline spotinst-cli:373]: Why did you delete the !BOTH statement?

**Review by orarnon — CHANGES_REQUESTED** [review]

File: spotinst-cli [hunk 2]
```diff
@@ -227,6 +227,7 @@ parser.add_option("--scale-up", action="store", dest="scale_up", type=int, help=
 parser.add_option("--scale-down", action="store", dest="scale_down", type=int, help="scale down group by X number of instances")
 parser.add_option("--suspend", action="store", dest="suspend", choices=['AUTO_SCALE', 'AUTO_HEALING' , 'BOTH'], help="suspend scaling or healing for a group. (AUTO_SCALE/AUTO_HEALING/BOTH)")
 parser.add_option("--unsuspend", action="store", dest="unsuspend", choices=['AUTO_SCALE', 'AUTO_HEALING' , 'BOTH'], help="unsuspend scaling or healing for a group. (AUTO_SCALE/AUTO_HEALING/BOTH)")
+parser.add_option("--ttl", action="store", dest="ttl", type=int, help="set an expiration for your suspension request - in minutes")
```
Comment (noym94) [inline spotinst-cli]: @arnoldyahad add an example as well, so people won't use "5m" or something

File: spotinst-cli [hunk 3]
```diff
@@ -464,8 +463,12 @@ for group, group_values in groups_to_update.iteritems():
     print message
     res = query_api(req_path, token, req_type, payload)
     if res.status_code == 200:
-        if req_type == 'GET':
+        if req_type == 'GET' and options.show_response == 'True':
+            print res.content
+    if res.status_code == 200:
+        if req_type == 'POST' and options.show_response == 'True':
             print res.content
```
Comment (noym94) [inline spotinst-cli:488]: why do we need to print "res" and "res.content"?

Comment (arnoldyahad) [reply spotinst-cli:488]: we actually dont need res ill remove it
res.content shows the content of the response

File: spotinst-cli (hunk 1 above)
Comment (arnoldyahad) [reply spotinst-cli:373]: BOTH was relevant when there was only auto scale & auto heal
now the api provides much more 
https://api.spotinst.com/spotinst-api/elastigroup/amazon-web-services/suspend-processes/
`
AUTO_SCALE, AUTO_HEALING, OUT_OF_STRATEGY, PREVENTIVE_REPLACEMENT, REVERT_PREFERRED
`

Comment (orarnon) [reply spotinst-cli:373]: I would leave BOTH for now since this is our most used use-case.

**Comment by orarnon** [comment]
I'd add list-suspended after each suspend\un-suspend operation that can serve as a validator and visibility.
https://api.spotinst.com/spotinst-api/elastigroup/amazon-web-services/list-suspended-processes/

File: spotinst-cli [hunk 4]
```diff
@@ -225,8 +225,9 @@ parser.add_option("--target", action="store", dest="target", type=int, help="upd
 parser.add_option("--max", action="store", dest="max", type=int, help="update group maximum capacity, must supply with min and target")
 parser.add_option("--scale-up", action="store", dest="scale_up", type=int, help="scale up group by X number of instances")
 parser.add_option("--scale-down", action="store", dest="scale_down", type=int, help="scale down group by X number of instances")
-parser.add_option("--suspend", action="store", dest="suspend", choices=['AUTO_SCALE', 'AUTO_HEALING' , 'BOTH'], help="suspend scaling or healing for a group. (AUTO_SCALE/AUTO_HEALING/BOTH)")
+parser.add_option("--suspend", action="store", dest="suspend", choices=['AUTO_SCALE', 'AUTO_HEALING' , 'OUT_OF_STRATEGY', 'PREVENTIVE_REPLACEMENT' , 'REVERT_PREFERRED'], help="suspend scaling or healing for a group. (AUTO_SCALE/AUTO_HEALING/BOTH)")
```
Comment (ghost) [inline spotinst-cli]: Please add new suspend options to help string as well. Also, an explanation for BOTH option.

**Comment by arnoldyahad** [comment]
@orarnon when using the --show-response=True option you will actually see the response which returns what is already suspended, here is an example
`~/git_work/spotinst-cli/spotinst-cli -g staging-rv-bidder-hap --suspend=AUTO_SCALE --ttl=40 --show-response=True

Querying spotinst API, hold on...

###################################
    Found the following groups:
###################################

+---+----------------------------------------------+--------------+-----+--------+-----+
| # | Group name                                   | ID           | Min | Target | Max |
+---+----------------------------------------------+--------------+-----+--------+-----+
| 1 | staging-rv-bidder-haproxy.sonic-us.us-east-1 | sig-8d3eab57 | 2   | 2      | 4   |
+---+----------------------------------------------+--------------+-----+--------+-----+

Are you sure you want to updated these groups[y/n]? y

============================================
staging-rv-bidder-haproxy.sonic-us.us-east-1
============================================
sending request to suspend the group
{
  "request": {
    "id": "55584e47-e8b8-459c-bef8-77542fbf3053",
    "url": "/aws/ec2/group/sig-8d3eab57/suspension",
    "method": "POST",
    "timestamp": "2019-10-10T15:14:27.796Z"
  },
  "response": {
    "status": {
      "code": 200,
      "message": "OK"
    },
    "kind": "spotinst:aws:ec2:suspension",
    "items": [
      {
        "groupId": "sig-8d3eab57",
        "processes": [
          "AUTO_HEALING",
          "PREVENTIVE_REPLACEMENT",
          "AUTO_SCALE"
        ],
        "suspensions": [
          {
            "name": "AUTO_HEALING",
            "expiresAt": "2019-10-10T15:51:12.000Z"
          },
          {
            "name": "PREVENTIVE_REPLACEMENT",
            "expiresAt": "2019-10-10T15:53:27.000Z"
          },
          {
            "name": "AUTO_SCALE",
            "expiresAt": "2019-10-10T15:54:28.000Z"
          }
        ]
      }
    ],
    "count": 1
  }
}`

i agree we can do some cool json tables, its kinda out of scope but ill be glad to do it
wdyt ? @orarnon

File: spotinst-cli (hunk 4 above)
Comment (noym94) [reply spotinst-cli]: add more explanation for the "help"

File: spotinst-cli (hunk 1 above)
Comment (noym94) [reply spotinst-cli:373]: so... it's still deleted?

**Review by noym94 — APPROVED** [review]

File: spotinst-cli (hunk 1 above)
Comment (arnoldyahad) [reply spotinst-cli:373]: nope, its back :) 
because it is our most important usage

File: spotinst-cli (hunk 4 above)
Comment (arnoldyahad) [reply spotinst-cli]: done
